### PR TITLE
Upgrade python to 3.8

### PR DESCRIPTION
**Pull request by fulder**
Upgrade plugins (with no opened PRs) from old python 3.7 version to 3.8 and add the `python3` packages for easier plugin maintenance (where python3 binaries can be used instead of upgrading them for every minor python release)

Closes: https://github.com/ix-plugin-hub/iocage-plugin-index/issues/230

-------

Thank you for your submission to the FreeNAS community plugins repository!

Please ensure the following guidelines are met when submitting a new Plugin.

1. Add the entry to INDEX in alphabetical order
2. Includes a new icon in the ./icon/ directory
3. Ensure the submitted icon file is a valid PNG that is 128x128 in size
4. Add new plugin cirrus task

### CODE REVIEW

File: homeassistant.json
Comment (tprelog) [inline homeassistant.json:33]: Home Assistant has specific requirements for the Python version that is used.

- https://github.com/home-assistant/architecture/blob/master/adr/0002-minimum-supported-python-version.md
- https://github.com/home-assistant/architecture/blob/master/adr/0016-home-assistant-core.md

Also, the virtualenv needs to be reinstalled when Python is updated. I will need to refactor some of the plugin for this change to be automagically handled correctly. I've considered rebuilding the virtualenv every time a user runs the plugin update, however some users have reported this can take almost an hour. It really depends on the NAS hardware and number of integrations a user has added to their Home Assistant.

Comment (fulder) [reply homeassistant.json:33]: Thanks for the detailed explanation @tprelog. I added the `python3` package here (and in other plugin manifests) in order to make it easier for the maintainers to be able to support minor updates without need to changing anything at the plugin level. But realize it can be problematic when it comes to the 2 supported python versions you're describing and the homeassistant plugin (currently 3.9 and 3.8 if I understand it correctly: https://github.com/home-assistant/core/blob/d829df332d8353661268f45300ac919dd8eb5116/homeassistant/const.py#L11 ?).

During the bump I was checking the `post_install` scripts for the plugins to see if they are using the `python3` binary or e.g. already supporting the current `python3.8`. That said, this change should probably work right now (as the `python3` package is currently `python3.8`), but the issue is if it gets updated too fast when homeassistant core is not supporting the new one yet, do I understand this correctly? If so, this would mean this package shouldn't be used by the plugin post install anyways and shouldn't be needed for this plugin (in this case I can remove it again).

Comment (fulder) [reply homeassistant.json]: I've removed the package again. Better to not have it there if it can cause more problems than it would "solve".

Comment (tprelog) [reply homeassistant.json]: So yea, Home Assistant currently supports python 3.8 and 3.9. When the next minor version of Python is released support for 3.8 will be dropped. However it can take a few  months before the new version is fully supported.

There are other minor nuance as well. One example, when the plugin was using python 3.7, there was a package `py37-av` but `py38-av` is not available.

In the perfect world, we would see a method for implementing a plugin branch (or version) as we previously discussed several months ago (I just now realized I somehow missed the notifications [when you had tagged me](https://github.com/iocage/iocage/pull/1248), sorry about that)

I think Home Assistant is an edge case that really pushes the boundaries of what is currently possible within the iocage plugin system and a perfect example of why most experienced users would still recommend a standard jail over plugins but that discussion is off topic here.

**Review by sonicaj — COMMENTED** [review]
@fulder can you please confirm if we are not regressing for `esphome` ?

File: homeassistant.json
Comment (fulder) [reply homeassistant.json]: Hard and locked dependencies like that are never fun to work with :/ Good job maintaining this plugin even though it is this complex :+1: 

No problem about the versioning PRs/discussions in the `iocage` repo, reviews and comments are always much appreciated when you get some free time, but no stress at all :).

I will resolve this discussion as this PR won't change anything with Home Assistant plugin, feel free to open it up again if there's something more to discuss here.

**Comment by fulder** [comment]
@sonicaj, from my understanding `esphome` should be ready for this change. The current cirrus task is failing due to the plugin being hardcoded to the `python3.7` version. Once this is merge, the plugin PR here should be merged as well: https://github.com/tprelog/iocage-esphome/pull/7 or is there something I've missed regarding this plugin (like Home Assistant) @tprelog (I'm guessing it is fine because of your comment in the PR, just doublechecking :smiley_cat: ) ?

**Review by sonicaj — APPROVED** [review]
Thank you for clarifying @fulder, let's wait for @tprelog to confirm/merge and we can merge this asap as well then :)

**Comment by tprelog** [comment]
@sonicaj - Let's go ahead and merge this.

I can merge @fulders pull request in the ESPHome repo as well

**Comment by sonicaj** [comment]
Awesome, thank you @tprelog for confirming

**Comment by tprelog** [comment]
I just merged @fulder pull request on my end for ESPHome

**Comment by fulder** [comment]
Thanks for the merges! :)

**Pull request merged**
